kasa/device.py

```diff
@@ -503,7 +503,9 @@ def _get_replacing_attr(
             return None
 
         for attr in attrs:
-            if hasattr(check, attr):
+            # Use dir() as opposed to hasattr() to avoid raising exceptions
+            # from properties
```

Comment:
This feels rather dirty, maybe there's something in `inspect` module that could be used instead of this?

Not that critical as this is in the deprecated info path, so your call if you want to touch it.

Reply:
I think `dir` is ok for what we're doing here and I think inspect would probably just return the same.